src/limits/MemoryLimitListener.cc

```diff
@@ -65,6 +65,7 @@ MemoryLimitListener::MemoryLimitListener(uint64_t memoryLimitKb)
             Arg(2)>MEMORY_LIMIT_MARGIN/2));
 }
 tracer::TraceAction MemoryLimitListener::handleMemoryAllocation(uint64_t allocatedMemoryKb){
+    if(memoryLimitKb_==0) return tracer::TraceAction::CONTINUE; // If there is no limit, we allow for allocation
```

Comment:
I think for consistency with `onExecuteEvent`, we should still call `setMemoryPeak`, even when there's no limit.
Maybe it's not necessary, but it's a smaller change and less likely to break something.

In that case we need to move this `memoryLimitKb_` check to the `if` here: https://github.com/sio2project/sio2jail/blob/ff16abc3a124f50dc6146c93284ebadea3272cb6/src/limits/MemoryLimitListener.cc#L78

similar to how it's done here:

https://github.com/sio2project/sio2jail/blob/ff16abc3a124f50dc6146c93284ebadea3272cb6/src/limits/MemoryLimitListener.cc#L134-L144